activerecord/lib/active_record/connection_adapters/mysql_adapter.rb

```diff
@@ -282,6 +282,10 @@ def cast_value(value)
               super
             end
           end
+
+          def has_precision?
+            precision || 0
```

Comment:
Am I wrong, or this will return 0 if precision is not present?

Reply:
If precision is not present, the behavior differs between MySQL and PostgreSQL.
In MySQL, it means no precision, truncates microseconds.
In PostgreSQL, it means no limit, never truncates microseconds.